src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -260,7 +260,7 @@ protected virtual IHostnameVerifier GetSSLHostnameVerifier (HttpsURLConnection c
 				URL java_url = new URL (EncodeUrl (redirectState.NewUrl));
 				URLConnection java_connection;
 				if (UseProxy)
-					java_connection = java_url.OpenConnection ();
+					java_connection = java_url.OpenConnection (await GetJavaProxy (redirectState.NewUrl, cancellationToken));
```

Comment:
Does this constitute a "redirect"?  Do we need to worry about Authentication headers here?

Reply:
It doesn't constitute a redirect per se, look at the code a few lines above:

```csharp
var redirectState = new RequestRedirectionState {
    NewUrl = request.RequestUri,
    RedirectCounter = 0,
    Method = request.Method
};
while (true) {
    URL java_url = new URL (EncodeUrl (redirectState.NewUrl));
```

**if** there is a redirect, the URL used here will reflect it, otherwise it's the request URL.

Regarding the authentication headers, we do need to worry about them but they have a [separate PR](https://github.com/xamarin/xamarin-android/pull/2388) (which timed out and is rebuilding now).